Bugzilla/Markdown.pm

```diff
@@ -0,0 +1,73 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Markdown;
+use 5.10.1;
+use Moo;
+
+use Encode;
+use Mojo::DOM;
+use HTML::Escape qw(escape_html);
+use List::MoreUtils qw(any);
+
+has 'markdown_parser' => (is => 'lazy');
+has 'bugzilla_shorthand' => (
+  is      => 'ro',
+  default => sub {
+    require Bugzilla::Template;
+    \&Bugzilla::Template::quoteUrls;
+  }
+);
+
+sub _build_markdown_parser {
+  if (Bugzilla->has_feature('alien_cmark')) {
+    require Bugzilla::Markdown::GFM;
+    require Bugzilla::Markdown::GFM::Parser;
+    return Bugzilla::Markdown::GFM::Parser->new({
+      hardbreaks    => 1,
+      validate_utf8 => 1,
+      extensions    => [qw( autolink tagfilter table strikethrough )],
+    });
+  }
+  else {
+    return undef;
+  }
+}
+
+sub render_html {
+  my ($self, $markdown, $bug, $comment, $user) = @_;
+  my $parser             = $self->markdown_parser;
+  my $bugzilla_shorthand = $self->bugzilla_shorthand;
+  my @valid_text_tags    = ('p', 'li', 'td');
+
+  if ($parser) {
+    my $html = decode('UTF-8', $parser->render_html($markdown));
+    my $dom = Mojo::DOM->new($html);
```

Comment:
For now, let's strip img tags (because they could be used as a DDoS, e.g. a few hundred img tags sent to a bug page could trigger an ip block),
and input, button, form, ... any other dangerous ones tagfilter doesn't already fix?
Note github appears to silently remove input too.

```perl
my @bad_tags = qw( form button input );
$dom->find( join(', ', @bad_tags) )->map('remove')
```